### PR TITLE
TSL: Introduce `node.toTexture()` and `rtt()`

### LINKED ISSUE #27508: WebGPURenderer: Support computing texture in TSL
### Description

Just like in the examples `webgpu_compute_points` and `webgpu_compute_particles_rain` would it be possible to use the TSL Syntax with `tslFn` to compute texture storage data? @sunag 

For example I started to try to port the `webgpu_compute_texture` example to tsl but it seems that `texture_storage_2d` is not supported yet or I don't know how to use it:
```js
const width = 512, height = 512;
const storageTexture = new StorageTexture( width, height );

const computeShaderFn = tslFn( ( [ storageTexture ] ) => {

	// https://www.shadertoy.com/view/Xst3zN

	const posX = instanceIndex.mod( width );
	const posY = instanceIndex.div( height );
	const indexUV = vec2( posX, posY );

	const x = float( posX ).div( 50. );
	const y = float( posY ).div( 50. );

	const v1 = sin( x );
	const v2 = sin( y );
	const v3 = sin( x.add( y ) );
	const v4 = sin( sqrt( x.mul( x ).add( y.mul( y ) ) ).add( 5. ) );
	const v = v1.add( v2 ).add( v3 ).add( v4 );

	const PI = 3.14159265359;

	const r = sin( v );
	const g = sin( v.add( PI ) );
	const b = sin( v.add( PI ).sub( 0.5 ) );


	return textureStore( storageTexture, indexUV, vec4( r, g, b, 1 ) );

} );
const computeNode = computeShaderFn().compute( width * height );
```

Something to keep in mind would be to be able to, just like in #27367, have a WebGL fallback where we would somehow ping pong basic textures I guess.

### LINKED ISSUE #28754: WebGPURenderer: Make output of post-processing passes configurable.
### Description

Next to #28749, this is the second of two issues I have encountered when implementing the latest post-processing passes.

Modules like `AnamorphicNode` or `AfterImageNode` use a separate render pass to write their result to a `TextureNode` (pointing to a `renderTarget.texture`) that represents the input for the next pass. I did not adapt that pattern for `DepthOfFieldNode` since a subsequent pass in the chain does not _always_ require texture input. 

In the previous composer, color space conversion and tone mapping was implemented as a separate output pass. This is not required with the new renderer anymore since the node system is able to add the color space and tone mapping code to the final output node. This is comparable with uber shader generation since the system avoids a render pass by combining shader code.

The same should be possible for FX passes as well which perform just per-pixel operations. In fact, it is already possible in certain cases. E.g. you can chain a simple color adjustment like `vibrance()` after `dof()` without the need of an intermediate texture. 

Ideally, a pass (indirectly) tells its predecessor what type of input it needs. Consider this:
```js
postProcessing.outputNode = scenePassColor.pass1().pass2();
``` 
If `pass2` requires a texture, `pass1` returns an instance of `TextureNode` in its `setup()` method. If no texture input is required, it returns its main TSL function ( e.g. `anamorph()`, `afterImg()` , `sobel()` ) which returns a `vec4` representing the current pixel value.

Depending on how a pass should produce its result, different code should run in `updateBefore()`. If the result should be saved to a texture with a render pass, the code can look like in  `AnamorphicNode` or `AfterImageNode`. If no texture is required, `updateBefore()` can be simply update uniforms like in `DepthOfFieldNode`.

The last pass in a chain should always be configured with `vec4` since for outputting the result on screen via `QuadMesh` we don't need a separate render pass for producing a texture (in other words the `fragmentNode` of `QuadMesh.material` does not require a `TextureNode`).

### Solution

The hard part (the combination of shader code) is already in place thanks to the beautiful TSL and node material.

The remaining bit is some sort of automatic pass chain configuration. Each FX pass could get a new property that defines the input type (e.g. `pass.inputType = 'vec4' | 'texture'`). A new `analyze()` method checks the flag  and ensures each pass produces produces the correct input for its successor. 

@sunag I'm not sure yet where in the node system something like `analyze()` should be defined though. Do you have something in mind?

### Alternatives

Always rendering effects to intermediate texture nodes as input for the next pass. This will negatively affect performance though.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sunag**
Related issue: Closes #28754. Closes https://github.com/mrdoob/three.js/issues/27508.

**Description**

This PR simplifies the creation of procedural textures and solves the problem where sampler custom shading is required in Post Processing and Materials.

Example

https://raw.githack.com/sunag/three.js/dev-rtt-node/examples/webgpu_procedural_texture.html

```js
const uvScale = uniform( 4 );
const blurIntensity = uniform( .5 );

const procedural = checker( uv().mul( uvScale ) );
const proceduralToTexture = procedural.toTexture( 512, 512 ); // ( width, height ) <- texture size

material.colorNode = proceduralToTexture.gaussianBlur( blurIntensity, 10 );
```

image

### CODE REVIEW

File: examples/webgpu_postprocessing.html [hunk 1]
```diff
@@ -76,7 +76,7 @@
 				const dotScreenPass = scenePassColor.dotScreen();
 				dotScreenPass.scale.value = 0.3;
 
-				const rgbShiftPass = dotScreenPass.getTextureNode().rgbShift();
+				const rgbShiftPass = dotScreenPass.rgbShift();
```
Comment (Mugen87) [inline examples/webgpu_postprocessing.html:79]: Excited to see `getTextureNode()` is not required here anymore!

File: src/nodes/display/RGBShiftNode.js [hunk 2]
```diff
@@ -41,7 +41,7 @@ class RGBShiftNode extends TempNode {
 
 }
 
-export const rgbShift = ( node, amount, angle ) => nodeObject( new RGBShiftNode( nodeObject( node ), amount, angle ) );
+export const rgbShift = ( node, amount, angle ) => nodeObject( new RGBShiftNode( nodeObject( node ).toTexture(), amount, angle ) );
```
Comment (Mugen87) [inline src/nodes/display/RGBShiftNode.js:44]: >  nodeObject( node ).toTexture()

What if the node is already a texture node? Ideally, the conversion is not executed in this casa.

File: src/nodes/utils/RTTNode.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+	}
+
+}
+
+export default RTTNode;
+
+export const rtt = ( node, ...params ) => nodeObject( new RTTNode( nodeObject( node ), ...params ) );
+
+addNodeElement( 'toTexture', ( node, ...params ) => node.isTextureNode ? node : rtt( node, ...params ) );
```
Comment (Mugen87) [inline src/nodes/utils/RTTNode.js:104]: So `toTexture()` does not perform `rtt()` when the node is already a texture node. That's good! 😊

**Review by Mugen87 — APPROVED** [review]

**Comment by Mugen87** [comment]
Nit: We should try to not add build files to the PRs.

**Comment by mrdoob** [comment]
> ### 📦 Bundle size
> _Full ESM build, minified and gzipped._
> 
> Filesize `dev`	Filesize PR	Diff
> 682.2 kB (169 kB)	682.2 kB (169 kB)	+0 B
> ### 🌳 Bundle size after tree-shaking
> _Minimal build including a renderer, camera, empty scene, and dependencies._
> 
> Filesize `dev`	Filesize PR	Diff
> 459.4 kB (110.9 kB)	459.4 kB (110.9 kB)	+0 B

We should probably start tracking the size of `three.webgpu.js` here too 👀

**Comment by Mugen87** [comment]
> We should probably start tracking the size of three.webgpu.js here too 👀

@marcofugaro Can you help us this? There is now a separate three build with `WebGPURenderer` instead of `WebGLRenderer`. Ideally the report added in #25615 now honors both types of builds.

**Comment by marcofugaro** [comment]
Oh finally! Knew this day would come. Let's test tree-shaking for the `WebGPURenderer`!!

**Comment by sunag** [comment]
> Oh finally! Knew this day would come. Let's test tree-shaking for the WebGPURenderer!!

Thanks @marcofugaro, tree-shaking is still in progress, initially [here](https://github.com/mrdoob/three.js/pull/28328) but it would be great if we could test it to check the progress of the updates.

**Pull request merged**